keras/engine/data_adapter.py

```diff
@@ -815,15 +816,10 @@ def __init__(self,
 
     self._first_batch_size = int(tf.nest.flatten(peek)[0].shape[0])
 
-    def _get_dynamic_shape(t):
-      shape = t.shape
-      # Unknown number of dimensions, `as_list` cannot be called.
-      if shape.rank is None:
-        return shape
-      return tf.TensorShape([None for _ in shape.as_list()])
+    def _get_tensor_spec(t):
+      return type_spec._type_spec_from_value(t)._with_tensor_ranks_only()
```

Comment:
It looks like there's an equivalent TF public API method with some extra type conversions, so that change is straightforward. It doesn't look like relaxing the tensor shapes is required to make the tests pass.